ring/ring.go

```diff
@@ -286,14 +286,13 @@ func (r *Ring) starting(ctx context.Context) error {
 func (r *Ring) loop(ctx context.Context) error {
 	// Update the ring metrics at start of the main loop.
 	r.updateRingMetrics()
-	go func() {
-		// Start metrics update ticker to update the ring metrics.
-		ticker := time.NewTicker(10 * time.Second)
-		defer ticker.Stop()
 
+	metricChan := make(chan struct{}, 1)
+	go func() {
 		for {
 			select {
-			case <-ticker.C:
+			case <-metricChan:
+				time.After(time.Second) // debounce and avoid consuming too much CPU
```

Comment:
We shouldn't wait 1 sec if it elapsed more than 1s since the last update. I think we need something slightly smarter here.

Reply:
went with different solution